File: ReSharper.FSharp/src/Services.FSharp/src/Injected/FSharpInjectionProvider.cs
using JetBrains.DataFlow;
using JetBrains.Lifetimes;
using JetBrains.ProjectModel;
using JetBrains.ReSharper.Plugins.FSharp.Psi;
using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree;
using JetBrains.ReSharper.Psi;
using JetBrains.ReSharper.Psi.Caches;
using JetBrains.ReSharper.Psi.Impl.Shared.InjectedPsi;
using JetBrains.ReSharper.Psi.RegExp.ClrRegex;
using JetBrains.ReSharper.Psi.RegExp.ClrRegex.Tree;

namespace JetBrains.ReSharper.Plugins.FSharp.Services.Cs.Injected
{
  [SolutionComponent]
  public class FSharpInjectionProvider : LanguageInjectorProviderInLiteralsWithRangeMarkersBase<IClrRegularExpressionFile,
    FSharpToken, FSharpLiteralInjectionTarget>
  {
    public FSharpInjectionProvider(Lifetime lifetime, ISolution solution, IPersistentIndexManager persistentIndexManager,
      InjectionNodeProvidersViewer providersViewer, FSharpLiteralInjectionTarget injectionTargetLanguage) : base(
      lifetime, solution, persistentIndexManager, providersViewer, injectionTargetLanguage)
    {
    }

    public override string ProvidedInjectionID => "FsRegex";
    public override PsiLanguageType SupportedOriginalLanguage => FSharpLanguage.Instance;
    public override PsiLanguageType ProvidedLanguage => ClrRegexLanguage.Instance;
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Injected/FSharpInjectionProvider.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Injected

open JetBrains.Lifetimes
open JetBrains.ProjectModel
open JetBrains.ReSharper.Plugins.FSharp.Psi
open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
open JetBrains.ReSharper.Psi.Caches
open JetBrains.ReSharper.Psi.Impl.Shared.InjectedPsi
open JetBrains.ReSharper.Psi.RegExp.ClrRegex
open JetBrains.ReSharper.Psi.RegExp.ClrRegex.Tree

[<SolutionComponent>]
type FSharpInjectionProvider
        (lifetime: Lifetime, solution: ISolution, persistentIndexManager: IPersistentIndexManager,
         providersViewer: InjectionNodeProvidersViewer, injectionTargetLanguage: FSharpLiteralInjectionTarget) =
    inherit LanguageInjectorProviderInLiteralsWithRangeMarkersBase<IClrRegularExpressionFile, FSharpToken, FSharpLiteralInjectionTarget>(lifetime, solution, persistentIndexManager, providersViewer, injectionTargetLanguage)

    override __.ProvidedInjectionID = "FsRegex"
    override __.SupportedOriginalLanguage = FSharpLanguage.Instance :> _
    override __.ProvidedLanguage = ClrRegexLanguage.Instance :> _
